src/components/FilterSidebar/MobileFilterSidebar.tsx

```diff
@@ -0,0 +1,39 @@
+import { twMerge } from 'tailwind-merge'
+import CircleCloseButton from '../UI/Buttons/CircleCloseButton/CircleCloseButton'
+import { ReactNode, useEffect, useRef } from 'react'
+import { useOnClickOutside, useMediaQuery } from 'usehooks-ts'
+
+interface IMobileFilterSidebar {
+  readonly className?: string
+  readonly children: ReactNode
+  readonly onClose: () => void
+}
+
+export default function MobileFilterSidebar({ onClose, className, children }: IMobileFilterSidebar) {
+  const sidebarRef = useRef<HTMLElement>(null)
```

Comment:
You can replace "IMobileFilterSidebar" with "Readonly\<IMobileFilterSidebar\>" and remove "readonly" from interface. If you agree, you can change it here and in other places from this pull request